Formula/jank-git.rb

```diff
@@ -11,34 +11,143 @@ class Jank < Formula
 
   depends_on "boost"
   depends_on "libzip"
-  depends_on "llvm@21"
+  depends_on "llvm"
   depends_on "openssl"
 
   skip_clean "bin/jank"
 
   def install
-    ENV.prepend_path "PATH", Formula["llvm@21"].opt_bin
+    # Clear Nix-provided compiler flags that point to old SDK (11.3)
+    # This is critical when building on systems with Nix installed
+    # These variables interfere with proper SDK and libc++ header resolution
+    ENV.delete("NIX_CFLAGS_COMPILE")
+    ENV.delete("NIX_LDFLAGS")
+    ENV.delete("NIX_APPLE_SDK_VERSION")
+    ENV.delete("MACOSX_DEPLOYMENT_TARGET")
+    ENV.delete("SDKROOT") # Clear Nix's SDKROOT pointing to old SDK
 
-    ENV.append "LDFLAGS", "-Wl,-rpath,#{Formula["llvm@21"].opt_lib}"
-
-    ENV.append "CPPFLAGS", "-L#{Formula["llvm@21"].opt_include}"
-    ENV.append "CPPFLAGS", "-fno-sized-deallocation"
+    # Use Homebrew's LLVM (not system clang, not Nix clang)
+    llvm = Formula["llvm"]
+    ENV.prepend_path "PATH", llvm.opt_bin
+    ENV["CC"] = llvm.opt_bin/"clang"
+    ENV["CXX"] = llvm.opt_bin/"clang++"
 
+    # Use the full Xcode SDK for macOS headers/frameworks (not CommandLineTools)
+    # Homebrew's MacOS.sdk_path returns the CLT SDK, so query xcrun directly.
     if OS.mac?
-      ENV["SDKROOT"] = MacOS.sdk_path
-    else
-      ENV["CC"] = Formula["llvm@21"].opt_bin/"clang"
-      ENV["CXX"] = Formula["llvm@21"].opt_bin/"clang++"
+      sdk = Utils.safe_popen_read("/usr/bin/xcrun", "--sdk", "macosx", "--show-sdk-path").strip
+      developer_dir = Utils.safe_popen_read("/usr/bin/xcode-select", "--print-path").strip
+
+      ENV["SDKROOT"] = sdk
+      ENV["HOMEBREW_SDKROOT"] = sdk
+      ENV["DEVELOPER_DIR"] = developer_dir
+
+      # Patch CMakeLists so the generated JIT flags don't bake CommandLineTools paths.
+      inreplace "compiler+runtime/CMakeLists.txt" do |s|
```

Comment:
This feels quite fragile, in that it'd break if upstream re-orders some things or even changes whitespace. Is there a preferred way to support this instead?

Reply:
Removed!